sql-schema/270.sql

```diff
@@ -0,0 +1,11 @@
+ALTER TABLE devices ADD location_id int NULL AFTER location_id;
+ALTER TABLE locations MODIFY lat double(10,6);
+ALTER TABLE locations MODIFY lng double(10,6);
+
+DELETE t1 FROM locations t1 INNER JOIN locations t2 WHERE t1.id < t2.id AND t1.location = t2.location;
+INSERT INTO locations (location, timestamp) SELECT devices.location, NOW() FROM devices WHERE devices.location IS NOT NULL AND NOT EXISTS (SELECT location FROM locations WHERE location = devices.location);
+UPDATE devices INNER JOIN locations ON devices.location = locations.location SET devices.location_id = locations.id;
+ALTER TABLE devices DROP location;
```

Comment:
I know for us, this is going to break groups and I think removes the ability to use location for groups now as the `locations` table can't reference back to a device id.

Reply:
We could join the table for all queries.

I'll check alerts and groups and see what our options are.

Bleh, new this was going to be a tricky one.

Reply:
Fixed